kore/test/Test/ConsistentKore.hs

```diff
@@ -0,0 +1,849 @@
+        context <- Reader.ask
+        quantifiedVariable <- setVariableGen childSort
+        withContext (addQuantifiedSetVariable quantifiedVariable context) $ do
+            child <- childGenerator childSort
+            return (builder quantifiedVariable child)
+
+binaryFreeSortOperatorGenerator
+    :: (Sort -> TermLike Variable -> TermLike Variable -> TermLike Variable)
+    -> Gen TermGenerator
+binaryFreeSortOperatorGenerator builder = do
+    context <- Reader.ask
+    return TermGenerator
+        { arity = 2
+        , sort = AnySort
+        , generator = worker context
+        }
+  where
+    worker context childGenerator resultSort = withContext context $ do
+        childSort <- sortGen
+        child1 <- childGenerator childSort
+        child2 <- childGenerator childSort
+        return (builder resultSort child1 child2)
+
+binaryOperatorGenerator
+    :: (TermLike Variable -> TermLike Variable -> TermLike Variable)
+    -> Gen TermGenerator
+binaryOperatorGenerator builder = do
+    context <- Reader.ask
+    return TermGenerator
+        { arity = 2
+        , sort = AnySort
+        , generator = worker context
+        }
+  where
+    worker context childGenerator childSort =
+        builder
+            <$> withContext context (childGenerator childSort)
+            <*> withContext context (childGenerator childSort)
+
+andGenerator :: Gen TermGenerator
+andGenerator = binaryOperatorGenerator mkAnd
+
+bottomGenerator :: Gen TermGenerator
+bottomGenerator = nullaryFreeSortOperatorGenerator mkBottom
+
+ceilGenerator :: Gen TermGenerator
+ceilGenerator = unaryFreeSortOperatorGenerator mkCeil
+
+equalsGenerator :: Gen TermGenerator
+equalsGenerator = binaryFreeSortOperatorGenerator mkEquals
+
+existsGenerator :: Gen TermGenerator
+existsGenerator = unaryQuantifiedElementOperatorGenerator mkExists
+
+floorGenerator :: Gen TermGenerator
+floorGenerator = unaryFreeSortOperatorGenerator mkFloor
+
+forallGenerator :: Gen TermGenerator
+forallGenerator = unaryQuantifiedElementOperatorGenerator mkForall
+
+iffGenerator :: Gen TermGenerator
+iffGenerator = binaryOperatorGenerator mkIff
+
+impliesGenerator :: Gen TermGenerator
+impliesGenerator = binaryOperatorGenerator mkImplies
+
+inGenerator :: Gen TermGenerator
+inGenerator = binaryFreeSortOperatorGenerator mkIn
+
+muGenerator :: Gen TermGenerator
+muGenerator = muNuOperatorGenerator mkMu
+
+notGenerator :: Gen TermGenerator
+notGenerator = unaryOperatorGenerator mkNot
+
+nuGenerator :: Gen TermGenerator
+nuGenerator = muNuOperatorGenerator mkNu
+
+orGenerator :: Gen TermGenerator
+orGenerator = binaryOperatorGenerator mkOr
+
+rewritesGenerator :: Gen TermGenerator
+rewritesGenerator = binaryOperatorGenerator mkRewrites
+
+topGenerator :: Gen TermGenerator
+topGenerator = nullaryFreeSortOperatorGenerator mkTop
+
+evaluatedGenerator :: Gen TermGenerator
+evaluatedGenerator = unaryOperatorGenerator mkEvaluated
+
+maybeStringLiteralGenerator :: Gen (Maybe TermGenerator)
```

Comment:
I think it's a little awkward returning the `TermGenerator` inside `Gen` when we don't do any generation to prepare the `TermGenerator`. I would say the same thing about the builtin generators below.

Reply:
Fixed. Some of these functions got an extra argument, but that's ok, I guess.